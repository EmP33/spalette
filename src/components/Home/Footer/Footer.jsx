import React from "react";
import styles from "./Footer.module.scss";
import CSSModules from "react-css-modules";

const Footer = () => {
  return <div></div>;
};

export default CSSModules(Footer, styles);
