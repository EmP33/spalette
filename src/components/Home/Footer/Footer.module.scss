.footer {
  width: 100%;
  font-size: 2rem;
  display: flex;
  align-items: center;
  justify-content: space-between;
  text-align: center;
  &__ghprofile {
    @media screen and(min-width:600px){
      margin-left:1rem;
    }
  }
  &__ghrepo {
   @media screen and(min-width:600px){
      margin-right:1rem;
    }
  }
}
