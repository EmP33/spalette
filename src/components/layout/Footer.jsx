import React from "react";
import styles from "./Footer.module.scss";
import CSSModules from "react-css-modules";

const Footer = () => {
  return (
    <footer styleName="footer">
      <div>
        <a href="https://github.com/EmP33" target="_blank" rel="noreferrer">
          Made by EmP33
        </a>
      </div>
    </footer>
  );
};

export default CSSModules(Footer, styles);
