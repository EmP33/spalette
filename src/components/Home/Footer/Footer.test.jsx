import { shallow } from "enzyme";
import Footer from "./Footer";

describe("Footer Component", () => {
  const wrapper = shallow(<Footer backgroundColor="#111" colorText="#fff" />);
  it("should render Footer component with correct background and text color", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("elements inside footer and footer should have correct background and text color", () => {
    expect(wrapper.prop("style")).toHaveProperty("background", "#111");
    expect(wrapper.find(".footer__ghprofile").prop("style")).toHaveProperty(
      "color",
      "#fff"
    );
    expect(wrapper.find(".logo").prop("style")).toHaveProperty("color", "#fff");
    expect(wrapper.find(".footer__ghrepo").prop("style")).toHaveProperty(
      "color",
      "#fff"
    );
  });
});
