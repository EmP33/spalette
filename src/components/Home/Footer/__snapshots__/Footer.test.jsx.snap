// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer Component should render Footer component with correct background and text color 1`] = `
<div
  className="footer"
  style={
    Object {
      "background": "#111",
    }
  }
>
  <a
    className="footer__ghprofile"
    href="https://github.com/EmP33"
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Github Profile
  </a>
  <h3
    className="logo"
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Spalette
  </h3>
  <a
    className="footer__ghrepo"
    href="https://github.com/EmP33/spalette"
    style={
      Object {
        "color": "#fff",
      }
    }
  >
    Github Repo
  </a>
</div>
`;
